src/rust/engine/ui/src/console_ui.rs

```diff
@@ -199,24 +193,19 @@ impl ConsoleUI {
   ///
   /// If the ConsoleUI is running, completes it.
   ///
-  pub fn teardown(&mut self) -> impl Future<Output = Result<(), String>> {
+  pub fn teardown(&mut self) {
     if let Some(instance) = self.instance.take() {
       LOGGER.deregister_stderr_handler(instance.logger_handle);
-      instance
-        .multi_progress_task
-        .map_err(|e| format!("Failed to render UI: {}", e))
-        .boxed()
-    } else {
-      future::ok(()).boxed()
+      std::mem::drop(instance);
```

Comment:
Now that the `instance` doesn't hold onto a handle to the `MultiProgress`, it's not clear how we know that it has shut down. Is the idea that dropping the `ProgressBar`s will cause the `MP` to stop rendering?

Perhaps continuing to hold onto it, and calling `mp.join_and_clear()` here would avoid the need to sleep?